src/theory/strings/solver_state.cpp

```diff
@@ -200,6 +201,62 @@ bool SolverState::areDisequal(Node a, Node b) const
 
 eq::EqualityEngine* SolverState::getEqualityEngine() const { return &d_ee; }
 
+void SolverState::getDisequalityList(std::vector<Node>& diseqs) const
```

Comment:
Instead of copying the values to a vector, I think that we should just return a `const` reference to the `NodeList` here. Also note that `CDList` defines `begin()` and `end()`, so we can iterate over it using a range-based for loop, i.e. `for (const Node& n : d_eeDisequalities)`.

Reply:
Yeah you're right, I'll update.